src/lib/services/client-metrics/instance-service.test.ts

```diff
@@ -0,0 +1,221 @@
+import ClientInstanceService from './instance-service';
+import getLogger from '../../../test/fixtures/no-logger';
+import { IClientApp } from '../../types/model';
+import { secondsToMilliseconds } from 'date-fns';
+
+/**
+ * A utility to wait for any pending promises in the test subject code.
+ * For instance, if the test needs to wait for a timeout/interval handler,
+ * and that handler does something async, advancing the timers is not enough:
+ * We have to explicitly wait for the second promise.
+ * For more info, see https://stackoverflow.com/a/51045733/2868829
+ *
+ * Usage in test code after advancing timers, but before making assertions:
+ *
+ * test('hello', async () => {
+ *    jest.useFakeTimers('modern');
+ *
+ *    // Schedule a timeout with a callback that does something async
+ *    // before calling our spy
+ *    const spy = jest.fn();
+ *    setTimeout(async () => {
+ *        await Promise.resolve();
+ *        spy();
+ *    }, 1000);
+ *
+ *    expect(spy).not.toHaveBeenCalled();
+ *
+ *    jest.advanceTimersByTime(1500);
+ *    await flushPromises(); // this is required to make it work!
+ *
+ *    expect(spy).toHaveBeenCalledTimes(1);
+ *
+ *    jest.useRealTimers();
+ * });
+ */
+function flushPromises() {
+    return Promise.resolve(setImmediate);
+}
```

Comment:
Nice, then our metrics tests should not be to blame for jest complaining about hanging timers :)